File: Assignment_1/app/src/main/java/com/kaushiknsanji/learndagger/ui/home/HomeViewModel.java
package com.kaushiknsanji.learndagger.ui.home;

import com.kaushiknsanji.learndagger.data.local.DatabaseService;
import com.kaushiknsanji.learndagger.data.remote.NetworkService;
import com.kaushiknsanji.learndagger.di.scope.FragmentScope;
import com.kaushiknsanji.learndagger.utils.NetworkHelper;

import javax.inject.Inject;

/**
 * Dummy ViewModel class for the {@link HomeFragment}
 *
 * @author Kaushik N Sanji
 */
@FragmentScope
public class HomeViewModel {

    private DatabaseService databaseService;
    private NetworkService networkService;
    private NetworkHelper networkHelper;

    @Inject
    public HomeViewModel(DatabaseService databaseService, NetworkService networkService, NetworkHelper networkHelper) {
        this.databaseService = databaseService;
        this.networkService = networkService;
        this.networkHelper = networkHelper;
    }

    public String getSomeData() {
        return databaseService.getDummyData() + " : "
                + networkService.getDummyData()
                + "\nIs Network Connected? "
                + networkHelper.isNetworkConnected();
    }
}

File: Assignment_2/app/src/main/java/com/kaushiknsanji/learndagger/ui/home/HomeViewModel.kt
package com.kaushiknsanji.learndagger.ui.home

import com.kaushiknsanji.learndagger.data.local.DatabaseService
import com.kaushiknsanji.learndagger.data.remote.NetworkService
import com.kaushiknsanji.learndagger.di.FragmentScope
import com.kaushiknsanji.learndagger.utils.NetworkHelper
import javax.inject.Inject

/**
 * Dummy ViewModel class for the [HomeFragment]
 *
 * @author Kaushik N Sanji
 */
@FragmentScope
class HomeViewModel @Inject constructor(
    private val databaseService: DatabaseService,
    private val networkService: NetworkService,
    private val networkHelper: NetworkHelper
) {

    /**
     * Method that returns some dummy data
     *
     * @return String containing some dummy data
     */
    fun getSomeData(): String =
        "${databaseService.getDummyData()} : ${networkService.getDummyData()} : ${networkHelper.isNetworkConnected()}"

}